File: src/main/java/com/decisionhub/service/VoteService.java
package com.decisionhub.service;

import com.decisionhub.dto.request.VoteRequest;
import com.decisionhub.dto.response.VoteResponse;
import com.decisionhub.dto.response.VoteResultResponse;

public interface VoteService {

    VoteResponse castVote(Long userId, VoteRequest request);

    VoteResponse changeVote(Long userId, Long voteId, VoteRequest request);

    VoteResponse castAnonymousVote(VoteRequest request);

    VoteResultResponse getVoteResults(Long decisionId);
}

File: frontend/src/services/voteService.js
/**
 * DecisionHub - Collaborative Decision-Making & Community Polling Platform
 *
 * File: voteService.js
 * Architecture Tier: API Service Wrapper (Data Layer)
 * Path: frontend/src/services/voteService.js
 *
 * Purpose:
 *   Voting API service encapsulating ballot submission, user vote retrieval, and voting analysis metrics.
 */

import api from './api';

export const voteService = {
  castVote: async (voteData) => {
    const response = await api.post('/votes', voteData);
    return response.data;
  },

  getVoteResults: async (decisionId) => {
    const response = await api.get(`/votes/result/${decisionId}`);
    return response.data;
  },
};
